src/baustein.js

```diff
@@ -841,6 +825,43 @@ var STATE_DESTROYING = 3;
 var STATE_DESTROYED = 4;
 
 /**
+ * Reducer function that can be used to turn a list of attributes into a JS object where the
+ * attribute names map to attribute values.
+ * @param {object} result
+ * @param {Attr} attr A DOM node attribute
+ * @returns {object}
+ */
+function attributeReducer(result, attr) {
+    result = result || {};
+    result[attr.name] = attr.value;
+    return result;
+}
+
+/**
+ * Copy all attributes from `source` to `target` and remove any attributes from `target` that are
+ * not present on `source`. The data-component-id attribute is ignored.
+ * @param {HTMLElement} target
+ * @param {HTMLElement} source
+ */
+function copyAttributes(target, source) {
```

Comment:
I'm not getting this function at all, it seems to be just a roundabout way to clone an object.